PyInstaller/loader/rthooks/pyi_rth_osgeo.py

```diff
@@ -20,3 +20,7 @@
 
 if os.path.exists(gdal_data):
     os.environ['GDAL_DATA'] = gdal_data
+
+# We *must* import site so that the 'help' builtin is added for Anaconda
```

Comment:
It is the only environment that I saw in the issue of having this problem, but based on what you've shown it does look like this must be for everything.